crates/provider/src/docker/client.rs

```diff
@@ -0,0 +1,444 @@
+    }
+
+    pub async fn container_rm(&self, name: &str) -> Result<()> {
+        let result = tokio::process::Command::new("docker")
+            .args(["rm", "--force", "--volumes", name])
+            .output()
+            .await
+            .map_err(|err| anyhow!("Failed do remove container '{name}: {err}"))?;
+
+        if !result.status.success() {
+            return Err(anyhow!(
+                "Failed to remove container '{name}': {err}",
+                err = String::from_utf8_lossy(&result.stderr)
+            )
+            .into());
+        }
+
+        Ok(())
+    }
+
+    pub async fn namespaced_containers_rm(&self, namespace: &str) -> Result<()> {
+        let container_names: Vec<String> = self
+            .get_containers()
+            .await?
+            .into_iter()
+            .filter_map(|container| match container {
+                Container::Docker(container) => {
+                    if let Some(name) = container.names.first() {
+                        if name.starts_with(namespace) {
+                            return Some(name.to_string());
+                        }
+                    }
+
+                    None
+                },
+                Container::Podman(container) => {
+                    if let Some(name) = container.names.first() {
+                        if name.starts_with(namespace) {
+                            return Some(name.to_string());
+                        }
+                    }
+
+                    None
+                },
+            })
+            .collect();
+
+        for name in container_names {
+            self.container_rm(&name).await?;
+        }
+
+        Ok(())
+    }
+
+    async fn get_containers(&self) -> Result<Vec<Container>> {
+        let containers = if self.using_podman {
+            self.get_podman_containers()
+                .await?
+                .into_iter()
+                .map(Container::Podman)
+                .collect()
+        } else {
+            self.get_docker_containers()
+                .await?
+                .into_iter()
+                .map(Container::Docker)
+                .collect()
+        };
+
+        Ok(containers)
+    }
+
+    async fn get_podman_containers(&self) -> Result<Vec<PodmanContainer>> {
+        let res = tokio::process::Command::new("docker")
+            .args(vec!["ps", "--all", "--no-trunc", "--format", "json"])
+            .output()
+            .await
+            .map_err(|err| anyhow!("Failed to get podman containers output: {err}"))?;
+
+        let stdout = String::from_utf8_lossy(&res.stdout);
+
+        let containers = serde_json::from_str(&stdout)
+            .map_err(|err| anyhow!("Failed to parse podman containers output: {err}"))?;
+
+        Ok(containers)
+    }
+
+    async fn get_docker_containers(&self) -> Result<Vec<DockerContainer>> {
+        let res = tokio::process::Command::new("docker")
+            .args(vec!["ps", "--all", "--no-trunc", "--format", "json"])
```

Comment:
We should add some filter to get only the containers of the `ns`
`--filter name=<ns>` will work